### PR TITLE
[cleanup][broker][Modernizer] fix violations in pulsar-broker

### PR DESCRIPTION
**Pull request by youzipi**
Master Issue: #12271 #16991 

### Motivation

Apply Maven Modernizer plugin to enforce we move away from legacy APIs.

### Modifications

- fix violations in pulsar-broker package.

### Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [x] `doc-not-needed` 

### Matching PR in forked repository

PR in forked repository: https://github.com/youzipi/pulsar/pull/1

### CODE REVIEW

**Comment by youzipi** [comment]
@MarvinCai @tisonkun

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java [hunk 1]
```diff
@@ -18,13 +18,13 @@
...
 package org.apache.bookkeeper.mledger;
 
-import com.google.common.base.Predicate;
 import com.google.common.collect.Range;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
```
Comment (tisonkun) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java:27]: Interesting. As we enable Modernizer for managed-ledger already, doesn't this file cause failure already?

**Review by tisonkun — APPROVED** [review]
LGTM. One comment out of curiosity.

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java (hunk 1 above)
Comment (youzipi) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java:27]: image
```
 /Users/mac/projects/source_code/pulsar/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription$3.java:-1: Prefer java.util.function.Predicate
```

here is the only one `guava Predicate` in PersistentSubscription, an anonymous class:
https://github.com/apache/pulsar/blob/e73b29863614a210d14e105c26f691c29d0548d5/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java#L558-L563

---
moderinzer use ASM to  scan the class files and detect the latency usage.

current rule only detect `implementing legacy interfaces`, see https://github.com/gaul/modernizer-maven-plugin/issues/1

I build a test case and debug it,
found that modernizer use a map to store all rules, and match with the class token by `absolutely equal`(not `contains`):

```java
private final Map<String, Violation> violations;
...
Violation violation = violations.get(token); 
checkToken(token, violation, name, lineNumber);
```

**Comment by youzipi** [comment]
@MarvinCai @codelipenghui

**Review by codelipenghui — APPROVED** [review]

**Review by lhotari — APPROVED** [review]

**Pull request merged**